### DISCUSSION TITLE
[AssetMapper] Dependencies not loaded for twigjs

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#52252] [Q&A]
Evidently [twigjs](https://www.jsdelivr.com/package/npm/twig) has some dependencies that I hope can be loaded automatically.

```bash
symfony new bug  --webapp --version=next --php=8.2 && cd bug
composer config extra.symfony.allow-contrib true
composer req symfony/asset-mapper:^6.4
bin/console importmap:require twig
bin/console make:controller Bug -i
cat > templates/bug.html.twig <<END
{% extends 'base.html.twig' %}
{% block javascripts %}
    {{ parent() }}
    <script type="module">
        import Twig from 'twig';
{% verbatim %}
        var template = Twig.twig({
            data: 'Hello, {{ name }}.'
        });
        document.getElementById("#hello").innerHTML = template.render({name: 'Tac'});
{% endverbatim %}
    </script>
{% endblock %}
{% block body %}<div id="hello">hello</div>{% endblock %}

END
symfony server:start -d
symfony open:local --path=/bug
```

Opening the console

![image](https://github.com/symfony/symfony/assets/619585/f4a6cd63-7e66-449f-9985-b95a510a0f6a)

importmap.php just has app and twig:

```php
return [
    'app' => [
        'path' => './assets/app.js',
        'entrypoint' => true,
    ],
    'twig' => [
        'version' => '1.16.0',
    ],
];
```

I imagine that this has to do with the way jsdelivr combines the packages, the top of loads the datetime and other packages.

https://cdn.jsdelivr.net/npm/twig@1.16.0/+esm

**Comment by weaverryan** [comment]
I'm on it - another new syntax for imports. I keep thinking we'll run out of new import syntaxes :)

**Comment by weaverryan** [comment]
Fixed in https://github.com/symfony/symfony/pull/52254

I tested and it works. However, with twig.js, you still get a warning in your console about the `require` function not being present (but the code works). The error comes from https://github.com/twigjs/twig.js/blob/master/src/twig.loader.fs.js#L9 - which I believe is not actually a relevant class when working in the browser (only if using Twig.js server-side via node). So, it'll be up to them, but that could, in theory, be an easy fix, but not sure.

Cheers!
